dora/core/client/fs/src/main/java/alluxio/client/file/dora/ConsistentHashPolicy.java

```diff
@@ -50,14 +53,18 @@ public ConsistentHashPolicy(AlluxioConfiguration conf) {
   }
 
   @Override
-  public List<BlockWorkerInfo> getPreferredWorkers(List<BlockWorkerInfo> blockWorkerInfos,
+  public List<BlockWorkerInfo> getPreferredWorkers(WorkerClusterView workerClusterView,
       String fileId, int count) throws ResourceExhaustedException {
-    if (blockWorkerInfos.size() < count) {
+    if (workerClusterView.size() < count) {
       throw new ResourceExhaustedException(String.format(
           "Not enough workers in the cluster %d workers in the cluster but %d required",
-          blockWorkerInfos.size(), count));
+          workerClusterView.size(), count));
     }
-    HASH_PROVIDER.refresh(blockWorkerInfos, mNumVirtualNodes);
+    List<BlockWorkerInfo> blockWorkerInfoList = workerClusterView.stream()
+        .map(w -> new BlockWorkerInfo(w.getIdentity(), w.getAddress(), w.getCapacityBytes(),
+            w.getUsedBytes(), w.getState() == WorkerState.LIVE))
```

Comment:
i thought by the time we get to this func, the original blockWorkerInfos is already set to be either a LIVE view or an ALL view

Reply:
oh its setting the mActive of BlockworkerInfo, never mind the above comment